packages/SwingSet/src/kernel/vatTranslator.js

```diff
@@ -79,12 +79,27 @@ function makeTranslateKernelDeliveryToVatDelivery(vatID, kernelKeeper) {
 
   function translateNotify(kpid, kp) {
```

Comment:
One of the things I discovered when I was writing the slogulator was that it turns out that lots of these translations change the shape of the thing being translated.  At first it bugged me and I was going to "fix" them, but the phenomenon was sufficiently widespread that changing the situation proved difficult and pointless and so I gave up.  So adding one more example of the phenomenon didn't seem to me to be incrementally worse.  The notify which does nothing (because the thing it was going to notify already has been) is a little whacky, but I don't see that it's a _problem_ per se that needs to be fixed.

Reply:
I like the shape we came up with in the meeting, let's see how that feels once it gets implemented. I kind of wish the translators were less mutating, but of course their primary purpose is to add entries to the clist when necessary. I do like the thing we came up with where the translator might add an entry, but never deletes them on its own; rather the non-vat-specific kernel code which creates the delivery object is responsible for any deletion that needs to happen afterwards.